src/org/benf/cfr/reader/bytecode/analysis/parse/expression/ArithmeticOperation.java

```diff
@@ -90,6 +92,174 @@ public Dumper dumpInner(Dumper d) {
         return d;
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        Literal l = lhs.getComputedLiteral(display);
+        if (l == null || !(l.getValue().getValue() instanceof Number))
+            return null;
+        Literal r = rhs.getComputedLiteral(display);
+        if (r == null || !(r.getValue().getValue() instanceof Number))
+            return null;
+        RawJavaType lType = l.getInferredJavaType().getRawType();
+        RawJavaType rType = r.getInferredJavaType().getRawType();
+        Number lv = ((Number) l.value.getValue());
+        Number rv = ((Number) r.value.getValue());
+        RawJavaType widestType = lType.ordinal() > rType.ordinal() ? lType : rType;
+        switch (widestType) {
+            case BOOLEAN:
+            case BYTE:
+            case CHAR:
+            case SHORT:
+            case INT:
+                return getComputedInt(rv.intValue(), lv.intValue());
+            case LONG:
+                return getComputedLong(rv.longValue(), lv.longValue());
+            case FLOAT:
+                return getComputedFloat(rv.floatValue(), lv.floatValue());
+            case DOUBLE:
+                return getComputedDouble(rv.doubleValue(), lv.doubleValue());
+        }
+        return null;
+    }
+
+    private Literal getComputedInt(int rv, int lv) {
+        int res;
+        switch (op) {
+            case PLUS:
+                res = lv + rv;
+                break;
+            case MINUS:
+                res = lv - rv;
+                break;
+            case MULTIPLY:
+                res = lv * rv;
+                break;
+            case DIVIDE:
+                res = lv / rv;
```

Comment:
This will make the decompiler fail with a / 0 error

Reply:
How should we handle these sorts of cases gracefully? Just ignore computation?